### PR TITLE
Refactor serialization_core_spec

### PR DESCRIPTION
**Pull request by shuheiktgw**
I refactored and cleaned up the test in `serialization_core_spec.rb` with `subject` and `let`. Also I added some missing tests! Thank you! 👍

### CODE REVIEW

File: spec/lib/serialization_core_spec.rb [hunk 1]
```diff
@@ -1,82 +1,212 @@
...
+        expect(id_hash[:type]).to eq 'movie'
       end
+    end
+
+    context 'when id is blank' do
+      let(:id) { nil }
 
-      result_hash = MovieSerializer.send(:id_hash, nil, 'movie')
-      expect(result_hash).to be nil
+      it 'returns nil' do
```
Comment (ankit8898) [inline spec/lib/serialization_core_spec.rb]: No need for `it 'returns nil'`,  `is_expected.to be_nil` will give the same message. Can use `it { is_expected.to be_nil }`

**Comment by shuheiktgw** [comment]
@ankit8898 Thank you so much for your comment! I squashed and modify the test you mentioned above! 👍

**Comment by shishirmk** [comment]
@ankit8898 The commits are squashed is it ok to merge?

File: spec/lib/serialization_core_spec.rb [hunk 2]
```diff
@@ -1,82 +1,210 @@
...
+    end
+  end
+
+  describe '#self.ids_hash_from_record_and_relationship' do
+    context 'when an association is not polymorphic' do
+      subject(:ids_hash) { MovieSerializer.ids_hash_from_record_and_relationship(movie, relationship) }
+
+      let(:relationship) { { name: :actors, relationship_type: :has_many, object_method_name: :actors, id_method_name: :actor_ids, record_type: :actor } }
+
+      it 'returns the correct hash' do
```
Comment (ankit8898) [inline spec/lib/serialization_core_spec.rb]: This returns Array. Can we update example desc?

File: spec/lib/serialization_core_spec.rb [hunk 3]
```diff
@@ -1,82 +1,210 @@
...
       end
     end
 
-    it 'returns the correct empty result when relationships_hash is called' do
-      movie.actor_ids = []
-      movie.owner_id = nil
-      relationships_hash = MovieSerializer.send(:relationships_hash, movie)
-      expect(relationships_hash[:actors][:data]).to eq([])
-      expect(relationships_hash[:owner][:data]).to eq(nil)
+    context 'when attribute without block' do
```
Comment (ankit8898) [inline spec/lib/serialization_core_spec.rb]: Should it be when attribute with block?

```ruby
context 'when attribute with block' do
 #...
end 
``` 
As you are passing a block to `MovieSerializer.attribute` ?

**Comment by ankit8898** [comment]
@shishirmk I missed submitting my additional review, realized after coming back from vacation. 😄 

@shuheiktgw Can you have a look at the additional one's? Mainly around example descriptions. Thanks!

**Comment by shishirmk** [comment]
Closing as we have no response

**Pull request closed without merging**